src/Runtime/CollectionTask.cs

```diff
@@ -82,7 +83,21 @@ internal async Task<string> StartCollection()
             _log.LogInformation("Consumers finished, closing output channel");
 
             foreach (var wkp in _context.LDAPUtils.GetWellKnownPrincipalOutput(_context.DomainName))
+            {
+                if (wkp.ObjectIdentifier.EndsWith(EnterpriseDCSuffix))
+                {
+                    if (wkp is Group g && g.Members.Length == 0)
+                    {
+                        continue;
+                    }
+                }
+                else
+                {
+                    wkp.Properties["reconcile"] = false;
+                }
```

Comment:
This could be rewritten to not be nested by doing:

```suggestion
                if (!wkp.ObjectIdentifier.EndsWith(EnterpriseDCSuffix))
                {
                    wkp.Properties["reconcile"] = false;
                }
                else  if (wkp is Group g && g.Members.Length == 0)
                {
                    continue;
                }
```